Sources/Slacket/Adapters/BodyParser.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  ParserType.swift
+//  Slacket
+//
+//  Created by Jakub Tomanik on 20/05/16.
+//
+//
+
+import Foundation
+import Kitura
+import SwiftyJSON
+import LoggerAPI
+
+extension ParsedBody {
+    
+    var isUTF8: Bool {
+        return true
+    }
+    
+    var contentType: String {
+        switch self {
+        case .text:
+            return "text/plain"
+        case .urlEncoded:
+            return "application/x-www-form-urlencoded"
+        case .json:
+            return "application/json"
+        case .multipart:
+            return "multipart/mixed"
+        case raw:
+            return "application/binary"
+        }
+    }
+    
+    var contentTypeHeaderKey: String {
+        return "Content-Type"
+    }
+    
+    var contentTypeHeaderValue: String {
+        var contentType = self.contentType
+        contentType += self.isUTF8 ? "; charset=utf-8" : ""
+        return contentType
+    }
+    
+    var header: [String: String] {
+        return [self.contentTypeHeaderKey: self.contentTypeHeaderValue]
+        
+    }
+    
+    var data: NSData? {
+        switch self {
+        case .text(let text):
+            return self.encode(text: text)
+        case .urlEncoded(let parameters):
+            return self.encode(parameters: parameters)
+        case .json(let json):
+            return self.encode(json: json)
+        case .multipart(let parts):
+            return self.encode(multipart: parts)
+        case .raw(let data):
+            return data
+        }
+    }
+    
+    init?(data: NSData, contentType contentTypeString: String?) {
+        guard let contentTypeString = contentTypeString else {
+            Log.debug("contentTypeString is nil")
+            return nil
+        }
+        
+        if contentTypeString.startsWith(prefix: "text/plain"), let text = String(data: data, encoding: NSUTF8StringEncoding) {
+            self = .text(text)
+        } else if contentTypeString.startsWith(prefix: "application/x-www-form-urlencoded"),
```

Comment:
have you considered switch case for this part? imo it would look cleaner, f.e. : 
`
switch contentTypeString {
case _ where contentTypeString.startsWith(prefix: .text.contentType):
...
}`